src/java/org/apache/cassandra/utils/TimeWindow.java

```diff
@@ -0,0 +1,94 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.cassandra.utils;
+
+import java.util.List;
+import java.util.Objects;
+
+/**
+ * Time window representing a range [ts, ts + duration). Use it with ms only to avoid any bugs.
```

Comment:
Maybe add ms to field names to make that explicit.

Reply:
Also would it make sense to use Interval instead. It's not in the start, duration style, but still..

Reply:
interval as in [start,end) instead of [start, start + duration)?

Reply:
Yes.